RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc

```diff
@@ -0,0 +1,730 @@
+  make_consumes(e_conf.clusterMaxDRDPhi	    , e_conf.tok_clusterMaxDRDPhi, sumes);
+  make_consumes(e_conf.clusterMaxDRDEta	    , e_conf.tok_clusterMaxDRDEta, sumes);
+  make_consumes(e_conf.clusterMaxDRRawEnergy, e_conf.tok_clusterMaxDRRawEnergy, sumes);
+  make_consumes(e_conf.clusterRawEnergy0    , e_conf.tok_clusterRawEnergy0, sumes);
+  make_consumes(e_conf.clusterRawEnergy1    , e_conf.tok_clusterRawEnergy1, sumes);
+  make_consumes(e_conf.clusterRawEnergy2    , e_conf.tok_clusterRawEnergy2, sumes);
+  make_consumes(e_conf.clusterDPhiToSeed0   , e_conf.tok_clusterDPhiToSeed0, sumes);
+  make_consumes(e_conf.clusterDPhiToSeed1   , e_conf.tok_clusterDPhiToSeed1, sumes);
+  make_consumes(e_conf.clusterDPhiToSeed2   , e_conf.tok_clusterDPhiToSeed2, sumes);
+  make_consumes(e_conf.clusterDEtaToSeed0   , e_conf.tok_clusterDEtaToSeed0, sumes);
+  make_consumes(e_conf.clusterDEtaToSeed1   , e_conf.tok_clusterDEtaToSeed1, sumes);
+  make_consumes(e_conf.clusterDEtaToSeed2   , e_conf.tok_clusterDEtaToSeed2, sumes);
+  make_int_consumes(e_conf.iPhi,      e_conf.tok_iPhi,sumes); 
+  make_int_consumes(e_conf.iEta,      e_conf.tok_iEta,sumes); 
+  make_consumes(e_conf.cryPhi,        e_conf.tok_cryPhi,sumes); 
+  make_consumes(e_conf.cryEta,        e_conf.tok_cryEta,sumes); 
+  
+  // setup photons 
+  if(!(empty_tag == ph_conf.photon_src)) 
+    ph_conf.tok_photon_src = sumes.consumes<edm::View<pat::Photon> >(ph_conf.photon_src);
+  
+  make_consumes(ph_conf.sigmaIetaIphi,ph_conf.tok_sigmaIetaIphi,sumes);
+  make_consumes(ph_conf.sigmaIphiIphi,ph_conf.tok_sigmaIphiIphi,sumes);
+  make_consumes(ph_conf.e2x5Max,ph_conf.tok_e2x5Max,sumes);
+  make_consumes(ph_conf.e2x5Left,ph_conf.tok_e2x5Left,sumes);
+  make_consumes(ph_conf.e2x5Right,ph_conf.tok_e2x5Right,sumes);
+  make_consumes(ph_conf.e2x5Top,ph_conf.tok_e2x5Top,sumes);
+  make_consumes(ph_conf.e2x5Bottom,ph_conf.tok_e2x5Bottom,sumes);
+  
+}
+
+template<typename T, typename U, typename V, typename Z>
+inline void assignValue(const T& ptr, const U& tok, const V& map, Z& value) {
+  if( !tok.isUninitialized() ) value = map.find(tok.index())->second->get(ptr.id(),ptr.key());
+}
+
+void EGExtraInfoModifierFromDB::modifyObject(pat::Electron& ele) const {
+  // we encounter two cases here, either we are running AOD -> MINIAOD
+  // and the value maps are to the reducedEG object, can use original object ptr
+  // or we are running MINIAOD->MINIAOD and we need to fetch the pat objects to reference
+  edm::Ptr<reco::Candidate> ptr(ele.originalObjectRef());
+  if( !e_conf.tok_electron_src.isUninitialized() ) {
+    auto key = eles_by_oop.find(ele.originalObjectRef().key());
+    if( key != eles_by_oop.end() ) {
+      ptr = key->second;
+    } else {
+      throw cms::Exception("BadElectronKey")
+        << "Original object pointer with key = " << ele.originalObjectRef().key() 
+        << " not found in cache!";
+    }
+  }
+
+  std::array<float, 33> eval;
```

Comment:
Hi Slava,
I am implementing the changes you requested.
What do you mean with this one  (or better what is the solution you are 
proposing) ?

On 8/8/15 22:53, Slava Krutelyov wrote:
>
> In RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc 
> <https://github.com/cms-sw/cmssw/pull/10613#discussion_r36581750>:
>
> > +  // we encounter two cases here, either we are running AOD -> MINIAOD
> > +  // and the value maps are to the reducedEG object, can use original object ptr
> > +  // or we are running MINIAOD->MINIAOD and we need to fetch the pat objects to reference
> > +  edm::Ptr<reco::Candidate> ptr(ele.originalObjectRef());
> > +  if( !e_conf.tok_electron_src.isUninitialized() ) {
> > +    auto key = eles_by_oop.find(ele.originalObjectRef().key());
> > +    if( key != eles_by_oop.end() ) {
> > +      ptr = key->second;
> > +    } else {
> > +      throw cms::Exception("BadElectronKey")
> > +        << "Original object pointer with key = " << ele.originalObjectRef().key()
> > +        << " not found in cache!";
> > +    }
> > +  }
> > +
> > +  std::array<float, 33> eval;
>
> the format better have been defined somewhere so that proper alignment 
> of parameters is more reliable to code
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/cms-sw/cmssw/pull/10613/files#r36581750>.
>

+++++++++++
Matteo Sani
University of California, San Diego, US.
Tel: +41227671577
CERN Office: 40-3-A02